File: app/src/main/java/com/mvvm/contactlist/db/dao/ContactDao.java
package com.mvvm.contactlist.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.mvvm.contactlist.db.entities.ContactEntity;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Single;

@Dao
public interface ContactDao {

    @Query("SELECT * FROM Tbl_Contact")
    Flowable<List<ContactEntity>> getAllContact();

    @Query("SELECT * FROM Tbl_Contact WHERE category_Id=:categoryId")
    Single<List<ContactEntity>> getContactFromCategory(int categoryId);

    @Query("SELECT * FROM Tbl_Contact WHERE contactId=:id")
    Maybe<ContactEntity> getContactFromContactId(int id);

    @Insert
    Completable insert(ContactEntity entity);

    @Update
    Completable update(ContactEntity entity);

    @Delete
    Completable delete(ContactEntity entity);
}

File: app/src/main/java/com/mvvm/contactlist/db/dao/ContactDao.kt
package com.mvvm.contactlist.db.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.mvvm.contactlist.db.entities.ContactEntity
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Maybe
import io.reactivex.Single

@Dao
interface ContactDao {

    @get:Query("SELECT * FROM Tbl_Contact")
    val allContact: Flowable<List<ContactEntity>>

    @Query("SELECT * FROM Tbl_Contact WHERE category_Id=:categoryId")
    fun getContactFromCategory(categoryId: Int): Single<List<ContactEntity>>

    @Query("SELECT * FROM Tbl_Contact WHERE contactId=:id")
    fun getContactFromContactId(id: Int): Maybe<ContactEntity>

    @Insert
    fun insert(entity: ContactEntity): Completable

    @Update
    fun update(entity: ContactEntity): Completable

    @Delete
    fun delete(entity: ContactEntity): Completable
}